### ISSUE TITLE
Indexing `tuple` which contains `TypeVarTuple` gives wrong type

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#5726]
**Describe the bug**
When indexing with a literal on a `tuple` which contains a `TypeVarTuple` results in the return type being incorrect.

**Code or Screenshots**
```python
from typing import TypeVar, TypeVarTuple

T = TypeVar("T")
Ts = TypeVarTuple("Ts")

def first(x: tuple[T, *Ts]) -> T:
    return x[0]
```
```py
/t.py:8:12 - error: Expression of type "T@first | Union[*Ts@first]" cannot be assigned to return type "T@first"
  Type "T@first | Union[*Ts@first]" cannot be assigned to type "T@first" (reportGeneralTypeIssues)
```

The type should be `T` but instead it is `Union[T, *Ts]`.

**VS Code extension or command-line**
`pyright 1.1.322`

**Comment by erictraut** [comment]
Yeah, pyright currently doesn't handle the case where the tuple is of indeterminate length. That's true for an unpacked TypeVarTuple or an unpacked `...` type (like `tuple[int, *tuple[str, ...]]`). These cases can be indeterminate, so it falls back on the more general solution that returns a union of all possible types.

It is possible to handle the limited case where a positive literal subscript targets a part of the tuple that is determinate. That's a reasonable feature request.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
